pkg/tools/dnsutils/dnsconfigs/tcp_dns_server_test.go

```diff
@@ -95,24 +101,39 @@ type udpHandler struct {
 }
 
 func (h *udpHandler) ServeDNS(rw dns.ResponseWriter, m *dns.Msg) {
-	if m.Question[0].Name == "my.domain." {
+	name := m.Question[0].Name
+
+	if name == health.ToWildcardPath(healthCheckHost) {
+		resp := prepareIPResponse(name, resolvedIP, m)
+		if err := rw.WriteMsg(resp); err != nil {
+			dns.HandleFailed(rw, m)
+		}
+		return
+	}
```

Comment:
Why do we need this?

Reply:
I did this to ensure that the server successfully resolved the health check hostname. When I get the expected response "health.check.only" -> "1.1.1.1" I know the server is ready

Reply:
I think we simply could use `memory` dns chain element with a one entry

Reply:
Great idea! Done